open_schools_platform/student_management/students/services.py

```diff
@@ -91,6 +92,15 @@ def create_studentprofileicrcle_additional(text: str = None, parent_phone: Phone
     return additional
 
 
+def get_student_profile_or_create_new(student_phone: str, student_name: str):
+    student_profile = get_student_profile(filters={"phone": student_phone})
+
+    if not student_profile:
+        student_profile = create_student_profile(name=student_name, age=0, phone=student_phone)
```

Comment:
it strange to create student_profile without name
partly this is a product error so ask product team for help